linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1371,8 +1371,12 @@ private void closeInner() {
       final Function2<TSource, TInner, TResult> resultSelector,
       boolean generateNullsOnLeft,
       boolean generateNullsOnRight) {
-    assert !generateNullsOnLeft : "not implemented";
-    assert !generateNullsOnRight : "not implemented";
+    if (generateNullsOnLeft) {
+      throw new UnsupportedOperationException("not implemented");
```

Comment:
Please use meaningful exception messages.

Do you really want to see `UnsupportedOperationException("not implemented")` in production logs? I bet you don't

Reply:
done